addons/html_editor/static/src/editor/table/table_plugin.js

```diff
@@ -284,15 +283,110 @@ export class TablePlugin extends Plugin {
+    //     if (targetNode.nodeType === Node.ELEMENT_NODE && isRow(targetNode) && !targetOffset) {
+    //         return true;
+    //     }
+    // }
+
+    /**
+     * Removes fully selected rows or columns, clears the content of selected
+     * cells otherwise.
+     *
+     * @param {NodeListOf<HTMLTableCellElement>} selectedTds - Non-empty
+     * NodeList of selected table cells.
+     */
+    deleteTableCells(selectedTds) {
+        const rows = [...closestElement(selectedTds[0], "tr").parentElement.children].filter(
+            (child) => child.nodeName === "TR"
+        );
+        const firstRowCells = [...rows[0].children].filter(
+            (child) => child.nodeName === "TD" || child.nodeName === "TH"
+        );
+        const firstCellRowIndex = getRowIndex(selectedTds[0]);
+        const firstCellColumnIndex = getColumnIndex(selectedTds[0]);
+        const lastCellRowIndex = getRowIndex(selectedTds[selectedTds.length - 1]);
+        const lastCellColumnIndex = getColumnIndex(selectedTds[selectedTds.length - 1]);
+
+        const areFullColumnsSelected =
+            firstCellRowIndex === 0 && lastCellRowIndex === rows.length - 1;
+        const areFullRowsSelected =
+            firstCellColumnIndex === 0 && lastCellColumnIndex === firstRowCells.length - 1;
+
+        if (areFullColumnsSelected) {
+            for (let index = firstCellColumnIndex; index <= lastCellColumnIndex; index++) {
+                this.removeColumn({ cell: firstRowCells[index] });
+            }
+            return;
+        }
+
+        if (areFullRowsSelected) {
+            for (let index = firstCellRowIndex; index <= lastCellRowIndex; index++) {
+                this.removeRow({ row: rows[index] });
+            }
+            return;
+        }
+
+        for (const td of selectedTds) {
+            // @todo @phoenix this replaces paragraphs by inline content. Is this intended?
+            td.replaceChildren(this.document.createElement("br"));
         }
+        this.shared.setCursorStart(selectedTds[0]);
     }
 
-    // @todo @phoenix to implement
-    deleteRangeBefore() {}
+    /**
+     * @param {Object} range - Range-like object.
+     * @param {Array} fullySelectedTables - Non-empty array of table elements.
+     */
+    deleteRangeWithFullySelectedTables(range, fullySelectedTables) {
+        let { startContainer, startOffset, endContainer, endOffset } = range;
+
+        // Expand range to fully include tables.
+        const firstTable = fullySelectedTables[0];
+        if (firstTable.contains(startContainer)) {
+            [startContainer, startOffset] = leftPos(firstTable);
+        }
+        const lastTable = fullySelectedTables.at(-1);
+        if (lastTable.contains(endContainer)) {
+            [endContainer, endOffset] = rightPos(lastTable);
+        }
+        range = { startContainer, startOffset, endContainer, endOffset };
+
+        let { cursorPos } = this.shared.deleteRange(range);
```

Comment:
This is wrong for nested tables

Reply:
I could not reproduce it, but I remember the table structure got broken after a few deleteBackwards. I have a test.todo for that.